controller/controller.go

```diff
@@ -240,12 +240,15 @@ func (c *Controller) deleteFunc(ctx context.Context, obj interface{}) {
 func (c *Controller) ProcessEvents(ctx context.Context, deleteChan chan watch.Event, updateChan chan watch.Event, errChan chan error) error {
 	loop := -1
 
+	// Memorise the original context to avoid growing the context forever.
```

Comment:
```suggestion
	// Memorise the original context to avoid growing the context forever as
	// we have seen issues in the past with this.
	//
	//     	https://github.com/giantswarm/giantswarm/issues/6416
	//
```